src/uk/gov/hmcts/contino/YarnBuilder.groovy

```diff
@@ -248,9 +248,20 @@ EOF
     return status == 0  // only a 0 return status is success
   }
 
+  private isYarnV1() {
+    def status = steps.sh label:"Determine if is YarnV1", script: '''
+                ! grep \"packageManager\": package.json
+          ''', returnStatus: true
+    return status
+  }
+
   def yarn(task) {
     if (!steps.fileExists(INSTALL_CHECK_FILE) && !runYarnQuiet("check")) {
-      runYarn("--mutex network install --frozen-lockfile")
+      if (!isYarnV1()) {
```

Comment:
A personal preference but if the positive case was first I find it easier to read, it could be something like:

```suggestion
      if (isYarnV2OrNewer()) {
```
or
flip the if statements around

there's some discussion around here:
https://softwareengineering.stackexchange.com/a/350551

which generally agree that positive is easier although not to do it at the expense of making it un-readable